packages/std/src/math/fraction.rs

```diff
@@ -12,3 +17,86 @@ pub trait Fraction<T>: Sized {
     /// If `p` is zero, None is returned.
     fn inv(&self) -> Option<Self>;
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct Fraction<T>(T, T);
+
+impl<T> Fraction<T> {
+    pub fn new(numerator: T, denominator: T) -> Self {
+        Self(numerator, denominator)
+    }
+}
+
+impl<T: Clone> Fractional<T> for Fraction<T> {
+    fn numerator(&self) -> T {
+        self.0.clone()
+    }
+
+    fn denominator(&self) -> T {
+        self.1.clone()
+    }
+
+    fn inv(&self) -> Option<Self> {
+        unimplemented!()
+    }
+}
+
+pub trait FractionMath {
```

Comment:
Do we need this trait? If we implement it that way, a user needs to imports `FractionMath` in order to use the types on Uint128. If we remove the trait and just have the macro, the functions are always available.